mbzirc_ign/src/GameLogicPlugin.cc

```diff
@@ -863,66 +829,7 @@ void GameLogicPlugin::PostUpdate(
   // triggers the start signal.
   if (!this->dataPtr->started)
   {
-    _ecm.Each<gazebo::components::Sensor,
-              gazebo::components::ParentEntity>(
-        [&](const gazebo::Entity &_entity,
-            const gazebo::components::Sensor *,
-            const gazebo::components::ParentEntity *_parent) -> bool
-        {
-          // Get the model. We are assuming that a sensor is attached to
-          // a link.
-          auto parent = _ecm.Component<gazebo::components::ParentEntity>(
-              _parent->Data());
-          auto model = parent;
-
-          // find top level model
-          while (parent && _ecm.Component<gazebo::components::Model>(
-                 parent->Data()))
-          {
-            model = parent;
-            parent = _ecm.Component<gazebo::components::ParentEntity>(
-                parent->Data());
-          }
-
-          if (model)
-          {
-            // \todo(anyone)
-            // Check if the robot has beyond the staging area.
-            // we need to trigger the /mbzirc/start.
-
-            // Get the model name
-            Entity entity = model->Data();
-            auto mName =
-              _ecm.Component<gazebo::components::Name>(entity);
-            if (this->dataPtr->robotNames.find(entity) ==
-                this->dataPtr->robotNames.end())
-            {
-              this->dataPtr->robotNames[entity] = mName->Data();
-              this->dataPtr->robotInitialPos[entity] =
-                  _ecm.Component<components::Pose>(entity)->Data().Pos();
-
-              // Subscribe to battery state in order to log battery events.
-              std::string batteryTopic = std::string("/model/") +
-                mName->Data() + "/battery/linear_battery/state";
-              this->dataPtr->node.Subscribe(batteryTopic,
-                  &GameLogicPluginPrivate::OnBatteryMsg, this->dataPtr.get());
-            }
```

Comment:
I think the above logic for logging dead battery events is missing after the refactor